agent/engine/engine_unix_integ_test.go

```diff
@@ -1331,3 +1369,40 @@ func TestTaskLevelVolume(t *testing.T) {
 	client := taskEngine.(*DockerTaskEngine).client
 	client.RemoveVolume(context.TODO(), "TestTaskLevelVolume", 5*time.Second)
 }
+
+func TestGPUAssociationTask(t *testing.T) {
+	gpuSupportEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_GPU_SUPPORT"), false)
+	if !gpuSupportEnabled {
+		t.Skip("Skipped because ECS_ENABLE_GPU_SUPPORT is not set to true")
+	}
+
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	client, err := sdkClient.NewClientWithOpts(sdkClient.WithHost(endpoint), sdkClient.WithVersion(sdkclientfactory.GetDefaultVersion().String()))
+	require.NoError(t, err, "Creating go docker client failed")
+
+	stateChangeEvents := taskEngine.StateChangeEvents()
+	testTask := createTestGPUTask()
+	container := testTask.Containers[0]
+
+	go taskEngine.AddTask(testTask)
+
+	verifyTaskIsRunning(stateChangeEvents, testTask)
+	assert.Equal(t, []string{"1"}, container.GPUIDs)
+	assert.Equal(t, "1", container.Environment[apitask.NvidiaVisibleDevicesEnvVar])
+
+	containerMap, _ := taskEngine.(*DockerTaskEngine).state.ContainerMapByArn(testTask.Arn)
+	cid := containerMap[testTask.Containers[0].Name].DockerID
+	state, _ := client.ContainerInspect(ctx, cid)
+	assert.Equal(t, apitask.NvidiaRuntime, state.HostConfig.Runtime)
+	assert.Contains(t, state.Config.Env, "NVIDIA_VISIBLE_DEVICES=1")
+
+	taskUpdate := *testTask
+	taskUpdate.SetDesiredStatus(apitaskstatus.TaskStopped)
+	go taskEngine.AddTask(&taskUpdate)
+	verifyTaskIsStopped(stateChangeEvents, testTask)
```

Comment:
Can we go a layer deeper?  Is there a command we can run within the container to validate that the gpu is present?

Reply:
Will definitely add this scenario in functional tests.